lib/net/dhcp/record.rb

```diff
@@ -52,10 +52,11 @@ def create
       end
     end
 
-    # Returns an array of record objects which are conflicting with our own
+    # Returns an array of record objects which are conflicting with our own. Comparsion is done
+    # only using selected attributes: MAC, IP, hostname and network.
     def conflicts
       conflicts = [proxy.record(network, mac), proxy.records_by_ip(network, ip)].flatten.compact.delete_if { |c| c.lease? || c == self || related_macs.include?(c.mac) }
-      @conflicts ||= conflicts.uniq {|c| c.attrs}
+      @conflicts ||= conflicts.uniq
```

Comment:
@ekohl has a point we do not need to rely on `==`. Could we keep that as is and do this instead?
```suggestion
      @conflicts ||= conflicts.uniq { |c| c.attrs.slice(:mac, :ip, :network) }
```

Reply:
@ezr-ondrej that doesn't work right, because the line above selects records and the `c == self` line does the comparison. That means you'll still end up with the same the entry while `@conflicts` should be empty when there are none.

Reply:
Ah, I see... so we would need to implement a method like this:
```ruby
def conflicts?(other)
  attrs.slice(:mac, :ip, :network) == other.attrs.slice(:mac, :ip, :network)
end
```
and use it, right? But I believe than the `==` operator would be useless :thinking:

Reply:
I dont understand, this completely drops the current behavior of skipping leases and related MACs.